### DISCUSSION TITLE
Is there a way to search for specific folders using regex inside a directory ?

### DISCUSSION THREAD

**Discussion by sailukar** [dnGrep/dnGrep#1238] [Q&A] [answered]
I have a directory structure as below

ParentDir
  dir1
  dir2
     interested-dir21
     dir22
  dir3
     pp-interested-dir31
     dp-interested-dir32
     dir3

![image](https://github.com/user-attachments/assets/9ccd1b87-0f6d-4766-af28-261631d601b1)

Iam interested to search only in the directories  that matches *interested* by providing the ParentDir in "Folder:", is this possible in dbgrep gui ?

**Comment by doug24** [comment, marked as answer]
You don't mention if there are files in ParentDir, dir2, or dir3 that you want to include or exclude.  If you are hoping to exclude any files in those three directories, then no, you can't do this starting from ParentDir.  If you exclude a directory from the search, then it excludes that directory and all of its children.  The only way to do that is to put three starting directories in the "Folder" field:

`c:\foo\Parent\dir2\interested-dir21;c:\foo\Parent\dir3\pp-interested-dir31;c:\foo\Parent\dp-interested-dir32`

If you are OK with searching files in ParentDir, dir2, or dir3 (or if there are no files in those directories), then you can exclude dir1, dir22, and dir3\dir3 by adding this to the "Patterns to exclude:" box:

`\dir1\;\dir22\;\dir3\dir3\`

Explanation: Exclude these are three directories, separated by semi-colon (;)
\dir1\
\dir22\
\dir3\dir3\
